libs/ui/src/hooks/use_smartcard_auth.ts

```diff
@@ -17,24 +16,72 @@ import {
   SuperadminLoggedInAuth,
   VoterLoggedInAuth,
   wrapException,
+  getBallotStyle,
+  getPrecinctById,
+  PrecinctSelectionKind,
+  PrecinctSelection,
+  VoterCardData,
+  VoterCardDataSchema,
+  Optional,
+  PrecinctId,
+  BallotStyleId,
+  CardlessVoterUser,
+  CardlessVoterLoggedInAuth,
 } from '@votingworks/types';
 import {
   assert,
   Card,
   CardApi,
   CardApiReady,
   throwIllegalValue,
+  utcTimestamp,
 } from '@votingworks/utils';
-import { useReducer } from 'react';
+import { useReducer, useState } from 'react';
 import useInterval from 'use-interval';
 import deepEqual from 'deep-eql';
 import { CARD_POLLING_INTERVAL } from './use_smartcard';
 import { Lock, useLock } from './use_lock';
 
-interface UseSmartcardAuthArgs {
-  cardApi: Card;
+export const VOTER_CARD_EXPIRATION_SECONDS = 60 * 60; // 1 hour
+
+// Below, we define some useful type guards for checking who's logged in.
```

Comment:
just moved these from the bottom to the top of the file so they can be used within the file